### DISCUSSION TITLE
Range for hovering around min-temp for heating

### DISCUSSION THREAD

**Discussion by xudabit** [revk/xESP32-Faikout#541] [Q&A]
Hello,

I'm using multiple faikin-modules with my ATXDs and I'm only using the faikin-auto-mode. My goal is to get one setting per unit for winter and summer to keep my automations as simple as possible (I have enough even without the ac-controls...). 

As an example I'm using something like this in my living-room:
autot: 23.0
autor: 2.0
pushtemp: 0.0

If the temperatur drops under 21.0° it starts with heating. When 21.0 or 21.1 is reached it stops with heating. This leads to the ac going on and off pretty often. I would like to extend the range for heating without raising the min threshold. First I thought "pushtemp" would be the right way, but this only raises the min-threshold and then it hovers around this threshold +-0.1° again. As a side question: Couldn't the same be accomplished with setting the target-temp a little higher and the range lower?

Is there any option to let the ac heat like from 21° to 21.5° and then stop until it lefts the range again (< 21.0°) with faikin-auto-mode? This would lead to longer running-periods with longer stop-periods and therefore should protect the unit and maybe save some energy. On the other hand this could decrease the comfort because of the oscilating temperature, but this would be personal preference.

**Comment by revk** [comment]
There is a thermostat setting which heats to top and drops to bottom and so on. This may be what you want.

**Reply by xudabit** [reply]
Thanks for your reply. If I'm using the settings from my question (autot: 23, autor: 2), wouldn't this mean, that it would start heating at 21 and stop heating at 25 and the other way around it would start cooling at 25 and cool down to 21? This would be way to cold in summer and way to hot in winter. If I would use something like 21+-0.5 it would be ok, but I would need to change the settings in summer because 21 is way to cold.

Summer and winter you might be able to change the target-temperature, but in the time where the outside-temperature is in between, you would be constantly switching modes, because sometimes you need to heat, sometimes you need to cool. And mostly it's ok if the temperature is a little outside without heating/cooling.

I mean more like a "small thermostat"-mode which only hovers around min (heating) and max (cooling). As far as I understand the code, in the check at line 3826 the min/max-values would need to be increased/decreased by this amount when it's not in thermostat-mode. Also the else-branch would need to get an if-statement to check if it's meant to be deactivated. Shouldn't this be sufficient to achieve such behavior? Surely there is more to do to add this as setting, make it adjustable via MQTT and so on. But sadly that seems to be way out of my knowledge with C...

**Reply by revk** [reply]
The 23±2 means it is "happy" anywhere in 21 to 25, so will happily stop heating at 21, or stop cooling at 25.

There is the `thermostat` setting to heat to 25 and then drop to 21, etc. Which I suspect is what you want.

Switching modes would be if outside range for some time.

There is also a setting to lock the mode and not switch between heat/cool automatically.

We have gone through a lot of ways people want it to work and there are settings to accommodate pretty much every option.

**Reply by xudabit** [reply]
The range is OK. Im fine with it being between 21° and 25°. 
Outside 23° -> Inside 23° -> Fine
Outside 29° -> Colling to 25° -> Fine
Outside < 10° -> Heating to 21° -> Fine
Outside < 10° -> Heating to 25° -> Way too hoot
Outside 29° -> Cooling to 21° -> Way too cold

So with thermostat-mode it would be way too hot or way too cold. Not sure if I'm the only on, but 25° in winter feel way different then 25° in summer. Normally I'm wearing thicker clothes in winter, so I don't want it that hot. In summer I'm wearing thinner clothes so I don't need it to be cooled that far down. That's what I want to achieve with the auto-mode and my range of +-2.0°. And that really works just fine.

My problem is, that it's hovering way to close around the 21° (+-0.1°). This leads to very small on-off-cycles which leads, as far as I know, to poor efficiency. The +-0.1° comes from the simple "<" or ">"-comparison which seems to be not enough, at least for me. 

It just checks if the actual temp (e.g. 21.1°) ist bigger than the min (21°) and it is. Now the ac will be turned off. The room is cooling down in minutes and reaches 20.9°. Now it's lower than 21° and it's heating again. At the current temperatures this happens multiple times an hour.

![image](https://github.com/user-attachments/assets/112e2951-2707-4f23-aa82-153cb2e20885)

I would like to sacrifice just a little comfort for better efficiency. My expectation would be a setting, just like "pushtemp", with lets the min as it is for "start heating", but raises the min for "end heating" (vice versa for cooling). So in my example it would start heating at 20.9° and stop heating at 21.5°. This would lead to way longer cycles and therefore should be more efficient.

With this option I would be able to keep the settings as they are in winter, summer and in between.

**Reply by revk** [reply]
I would suggest exploring the options, really. There a lot, even time settings to stop it switching rapidly.

**Reply by xudabit** [reply]
I really explored the options. But all have an unnecessary disadvantage. When I'm raising the timenoflap it could overheat the room way more than it should. This would be the case when it's not that cold outside and timenoflap is set pretty high. 

With thermostat-mode, autot at 21.3 and autor at 0.3 it seems to raise the cycles as I would expect. But, as I said, this leads to problems when the weather is changing. In the last two or three weeks we had temperatures here (western germany) from 5° to 25°. If I'm using thermostat-mode all the time, this would lead to cooling when we have 25° outside, which isn't necessary and would be a waste of energy/money.

**Reply by xudabit** [reply]
I tried some options yesterday. None of them were good for my use-case. The temperature was hold quiet well, but when I want to lower the temperature at night I have some problems.

Now I will try the thermostat-mode, because this seems to be at least half of what I would like to have. For heating I will use something like this:
autot: 21.2
autor: 0.3

I hope that this will do what I need.

If this works as intended, I will give it a try to implement what I was talking about for heating and cooling. In my opinion (at least in my case) min and max should behave like little thermostat-modes. When it's to cold, then heat for more than the default 0.1° and then stop and let it cool again until it's under min. Vice versa for cooling.

Would love to let the AC run longer cycles, but I'm not sure if this is currently possible. My Daikin ATXD/ARXD also seems to behave a little weird. Even with 4.5° difference (target and home) it sometimes starts up the fan on lowest setting.

By the way: I really appreciate your work with this. Just a little frustrating that I'm not able to get it work the way I intend to use it. And so far I don't see any option which lets me use it that way. Maybe my requirements are set to high.

**Comment by wurmtal868** [comment]
Regarding the weird behavior:
When in idle mode, the indoor unit will turn on the the fan every few minutes to avoid local air bubbles.
E.g. After a heating cycle there is still a lot energy in the liquid of the heat-exchanger which can result in a higher local temperature readings.
In the example below it took 3 extra fan cycles to bring the liquid temperature down to a steady state. The delta between room temperature and Daikin target is 9K.
![image](https://github.com/user-attachments/assets/ed7bd0fc-6753-4f4c-ab8f-b86d74563779)
